lib/modules/manager/pip-compile/common.ts

```diff
@@ -22,6 +23,11 @@ export function getPythonVersionConstraint(
     return python;
   }
 
+  if (extractedPythonVersion) {
+    logger.debug('Using python constraint extracted from the lock file');
+    return `==${extractedPythonVersion}`;
+  }
```

Comment:
pip-compile seems to embed a Python "stream" (e.g. 3.11) and not a version (e.g. 3.11.3). Shouldn't it be `==3.11.*` in such cases?

Reply:
I'm not sure how exactly Renovate resolves these constraints, but:

1) https://docs.renovatebot.com/modules/manager/pip-compile/#configuration-of-python-version shows `==3.7` as an example

2) In my repository I currently have `"constraints": { "python": "==3.12" }`. Then in the log:

```
DEBUG: Resolved stable matching version (branch="renovate/pytest-8.x")
{
  "toolName": "python"
  "constraint": "==3.12"
  "resolvedVersion": "3.12.3"
}
```

Also, one of the added tests verifies that for `Python 3.11` in the header, the latest version from 3.11 "stream" gets installed (`getPkgReleases` mocked to return `3.11.0` and `3.11.1`).

Reply:
No, I don't. Can create one if required - although tests in `artifacts.spec.ts` are almost "end to end".

Reply:
Thanks, I can confirm it worked for me in a fork of your reproduction